### PR TITLE
[ML][Pipelines] Bugfix: handle group during init/finalize validation

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Handle group during validate init/finalize, use `flatten()` to get all items and then validate.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py [hunk 1]
```diff
@@ -345,19 +345,30 @@ def _try_get_data_binding(_input_output_data) -> Union[str, None]:
                     return None
                 return _input_output_data._data_binding()
 
+            def _is_group_input(_input_obj) -> bool:
```
Comment (brynn-code) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:348]: How about GroupInput._is_group_attr_dict?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py [hunk 2]
```diff
@@ -345,19 +345,30 @@ def _try_get_data_binding(_input_output_data) -> Union[str, None]:
...
+                # TODO: update validate init/finalize logic after supporting group as output
+                return not hasattr(_input_obj, "_data")
+
             _validate_job = self.jobs[_validate_job_name]
             # no input to validate job
             for _input_name in _validate_job.inputs:
-                _data_binding = _try_get_data_binding(_validate_job.inputs[_input_name]._data)
+                _input = _validate_job.inputs[_input_name]
+                if _is_group_input(_input):
+                    continue
```
Comment (brynn-code) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:358]: Skip or `_input.flatten()` to get the inner inputs?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py [hunk 3]
```diff
@@ -345,19 +345,30 @@ def _try_get_data_binding(_input_output_data) -> Union[str, None]:
                     return None
                 return _input_output_data._data_binding()
 
+            def _is_group_input(_input_obj) -> bool:
+                # exclude group input for now as it only comes from pipeline input now
+                # TODO: update validate init/finalize logic after supporting group as output
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:350]: Please create a task under 1.4.0 to track this.

Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:350]: mmm.. we can also reuse the exiting bug

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py (hunk 1 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py]: Great, have replaced with this.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py (hunk 2 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py]: Add this in getting data binding function, use `flatten()` and pick first data binding for later use.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py (hunk 3 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py]: I have updated implementation and maybe this can be solved in this PR

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py [hunk 4]
```diff
@@ -337,31 +338,44 @@ def _is_control_flow_node(_validate_job_name: str) -> bool:
             return issubclass(type(_validate_job), ControlFlowNode)
 
         def _is_isolated_job(_validate_job_name: str) -> bool:
-            def _try_get_data_binding(_input_output_data) -> Union[str, None]:
-                """Try to get data binding from input/output data, return None if not found."""
+            def _try_get_data_bindings(_name: str, _input_output_data) -> Union[List[str], None]:
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:341]: add some test? may add in another PR

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py (hunk 4 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:341]: yes, and opened a task tracking this

**Review by brynn-code — APPROVED** [review]

**Pull request merged**